File: 9.palindrome-number.py
#
# @lc app=leetcode id=9 lang=python3
#
# [9] Palindrome Number
#

# @lc code=start
class Solution:
    def isPalindrome(self, x: int) -> bool:
        x_str = str(x)
        reversed_str = x_str[::-1]
        if x_str == reversed_str:
            return True
        else:
            return False
            
# @lc code=end

File: 9.palindrome-number.cpp
/*
 * @lc app=leetcode id=9 lang=cpp
 *
 * [9] Palindrome Number
 */

// @lc code=start
#include <iostream>
#include <string>
using namespace std;
class Solution {
public:
    bool isPalindrome(int x) {
        string x_str = to_string(x);
    string reversed_str = string(x_str.rbegin(), x_str.rend());
    if (x_str == reversed_str) {
        return true;
    } else {
        return false;
    }
    }
};
// @lc code=end

